Userland/Libraries/LibWeb/Layout/LineBox.h

```diff
@@ -20,7 +20,7 @@ class LineBox {
     CSSPixels bottom() const { return m_bottom; }
     CSSPixels baseline() const { return m_baseline; }
 
-    void add_fragment(Node const& layout_node, int start, int length, CSSPixels leading_size, CSSPixels trailing_size, CSSPixels leading_margin, CSSPixels trailing_margin, CSSPixels content_width, CSSPixels content_height, CSSPixels border_box_top, CSSPixels border_box_bottom, RefPtr<Gfx::GlyphRun> glyph_run = {});
+    void add_fragment(Node const& layout_node, int start, int length, CSSPixels leading_size, CSSPixels trailing_size, CSSPixels leading_margin, CSSPixels trailing_margin, CSSPixels content_width, CSSPixels content_height, CSSPixels border_box_top, CSSPixels border_box_bottom, RefPtr<Gfx::GlyphRun> glyph_run = {}, CSS::Direction direction = CSS::Direction::Ltr);
 
```

Comment:
This default is wrong. You should use the default direction of the document. For example in HTML4:

```
<html dir=rtl">
<body>
aa....!!!
```

The "!!!" should be at the left of the "aa".

Reply:
We now inherit the default direction, from the context box's computed CSS value.